bin/rule.ml

```diff
@@ -97,21 +97,23 @@ let pp_direction fmt = function
   | `To_md -> Fmt.pf fmt "--direction=to-md"
   | `To_ml -> Fmt.pf fmt "--direction=to-ml"
 
+let pp_section fmt s = Fmt.pf fmt "-s %S" s
+
 let pp_prelude fmt s = Fmt.pf fmt "--prelude=%s" s
 let pp_prelude_str fmt s = Fmt.pf fmt "--prelude-str %S" s
 
 let add_opt e s = match e with None -> s | Some e -> String.Set.add e s
 
 let run (`Setup ()) (`File md_file) (`Section section) (`Direction direction)
     (`Prelude prelude) (`Prelude_str prelude_str) (`Root root) =
-  let section = match section with
-    | None   -> None
-    | Some p -> Some (Re.Perl.compile_pat p)
-  in
-  let active b = match section, Mdx.Block.section b with
-    | None   , _      -> true
-    | Some re, None   -> Re.execp re ""
-    | Some re, Some s -> Re.execp re (snd s)
+  let active =
+    let active re b =
```

Comment:
I'm not sure it reads any better that way, I think it'd be clearer as a single pattern matching:
```ocaml
let active b =
  match section with
  | None -> true
  | Some p ->
   let re = Re.Perl.compile_pat p in
   let block_section = Util.Option.get_value ~default:"" (Mdx.Block.section b) in
   Re.execp re block_section
```

You can add the `Option.get_value` to the `Util` module, we'll probably need it again in the future!

Also I'm not sure the `-s` option behaviour when there's no section is really intuitive to users. It should probably be documented. That's unrelated to this PR but if you feel like adding that in a separate commit it'd be great!

Reply:
Correct me if I'm wrong but I feel like this contains a bug fix so it should probably sit in its own commit!

Reply:
I think I didn't change any behavior. What do you mean?

This code also unpack the tuple and I don't want to come up with a "get_value_map" function when a pattern matching is as easy to use.

Reply:
I originally changed the code to avoid shadowing `section`. Just changed so it's similar to how it was.

Reply:
Ok nevermind I misread the change, thought we change when the re was compiled but it's not the case!

I do prefer the old version so we'll move this within the `let active = ...` to prevent the shadowing of the `section` argument, as discussed offline!